crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -301,3 +355,48 @@ where
         }
     }
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Handle the epilogue error during the note consumption check in the `can_consume` method.
+///
+/// The goal of this helper function is to handle the cases where the account couldn't consume the
+/// note because of some epilogue check failure, e.g. absence of the authenticator.
+fn handle_epilogue_error(epilogue_error: TransactionExecutorError) -> NoteConsumptionStatus {
+    match epilogue_error {
+        // `Unauthorized` is returned for the multisig accounts if the
+        // transaction doesn't have enough signatures.
+        TransactionExecutorError::Unauthorized(_)
+        // `MissingAuthenticator` is returned for the account with the basic
+        // auth if the authenticator was not provided to the executor
+        // (UnreachableAuth).
+        | TransactionExecutorError::MissingAuthenticator => {
+            // Both these cases signal that there is a probability that the provided note could be
+            // consumed if the authentication is provided.
+            NoteConsumptionStatus::UnconsumableWithoutAuthorization
+        },
+        // TODO: apply additional checks to get the verbose error reason
+        _ => NoteConsumptionStatus::Undetermined,
+    }
+}
+
+// HELPER STRUCTURES
+// ================================================================================================
+
+#[derive(Debug, PartialEq)]
+pub enum NoteConsumptionStatus {
+    /// The note can be consumed by the account at the specified block height.
+    Consumable,
+    /// The note can be consumed by the account after the required block height is achieved.
+    /// TODO: use this variant during the static analysis of the well-known note.
+    ConsumableAfter(BlockNumber),
+    /// The note cannot be consumed by the account because the authorization is not provided.
+    /// TODO: return the expected authentication data
+    UnconsumableWithoutAuthorization,
+    /// The note cannot be consumed against the account, but the reason was not defined.
+    Undetermined,
+    /// The note cannot be consumed against the account with a reason.
+    /// TODO: add some value for this variant to hold the reason why the note cannot be consumed.
+    Unconsumable,
```

Comment:
Thinking more about the naming here, I wonder if we should change `Undetermined` -> `Unconsumable` and `Unconsumable` -> `Incompatible`. It could look something like:
```rust
/// The note cannot be consumed by the account at the specified conditions (i.e., block
/// height and account state).
Unconsumable,
/// The note cannot be consumed by the specified account under any conditions.
Incompatible,
```